### ISSUE TITLE
Overview page of all settings

### ISSUE THREAD

**Issue by mariobehling** [fossasia/accounts.susi.ai#125]
Provide an overview page of all settings and a specific section detailing settings for each client.

After the user logs in show this settings page.

**Comment by DravitLochan** [comment]
Something like this https://myaccount.google.com ?

**Comment by DravitLochan** [comment]
Should it be done on /userhome route or a new route is to be added?

**Comment by mariobehling** [comment]
Settings should be in a /settings route.

**Comment by DravitLochan** [comment]
@mariobehling Please tell for what all clients does this needs to be handled. If we are configuring for each client individually, then first a servlet needs to be implemented which stores settings for each client. Current one stores settings details in a generic one.

**Comment by DravitLochan** [comment]
@mariobehling I took motivation from our GSoC student dashboard and think that our settings dashboard would look good if we follow this with the SUSI theme. What do you think?
![susi_settings_sample](https://user-images.githubusercontent.com/14837478/28614079-1c2c3d9e-7212-11e7-8fc7-826f779a69d5.png)

**Comment by mariobehling** [comment]
Looks good. Please go ahead.

**Comment by mariobehling** [comment]
Please change the colors to the same as chat.susi.ai.

**Comment by chashmeetsingh** [comment]
Settings keys used in iOS:
- `enterToSend`: boolean
- 'micInput`: boolean
- 'hotwordRecognition`: boolean
- `speechOutput`: boolean
- 'theme`: light or dark

I can make changes if we are using different keywords on other clients as it's better to have common keys across all clients rather than client specific keys.

**Comment by amitiwary999** [comment]
Setting keys used in Android

enter_send: Boolean
speech_always: Boolean
speech_output: Boolean
mic_input: Boolean
hotword_detection: Boolean
theme: String

**Comment by DravitLochan** [comment]
@chashmeetsingh Right! @chiragw15 @amitiwary999 @madhavrathi @rishiraj824 Please coordinate on this and use same key-values for user settings. I will be implementing a servlet to delete existing user settings.

**Comment by chiragw15** [comment]
@chashmeetsingh @uday96 @amitiwary999 please follow the keys as discussed here https://github.com/fossasia/susi_android/pull/781#issuecomment-312442162 . Or if you want some change, do suggest it. Also, if possible, document it somewhere. It would be easier that way to follow up in future.

**Comment by chashmeetsingh** [comment]
@DravitLochan sure, go ahead remove the existing ones. Just a suggestion, can we assign default values to these keys and we can grab those default values directly from the server instead of having them on the clients? In case, in future, we think to make some modifications we can do them at one place and it takes effect at all the other places. Thoughts @chiragw15 @uday96

**Comment by rishiraj824** [comment]
@DravitLochan Yes the keys which we are following keys being followed for the settings in the web right now are :- 
```
                        theme: string,
			enterAsSend: boolean,
			micInput: boolean,
			speechOutput: boolean,
			speechOutputAlways: boolean
			speechRate: 0.1-5
			speechPitch: 0-2
```

**Comment by uday96** [comment]
Is this the finalised list then?

```
                        theme: string
			enterAsSend: boolean
			micInput: boolean
			speechOutput: boolean
			speechOutputAlways: boolean
			speechRate: 0.5-2
			speechPitch: 0-2
                        hotwordRecognition: boolean
                        ttsLanguage: string
                        prefLanguage: string
```

**Comment by chashmeetsingh** [comment]
yup, that's the final one.

**Comment by DravitLochan** [comment]
Ok. By tonight, I'll update here. One last thing. Please add the default values as well.

**Comment by uday96** [comment]
```
                        theme: 'light'
			enterAsSend:  true
			micInput: true
			speechOutput: true
			speechOutputAlways: false 
			speechRate: 1
			speechPitch: 1
                        hotwordRecognition: true? (not sure)  
                        ttsLanguage: 'en-US'
                        prefLanguage: 'en-US'
```

**Comment by chashmeetsingh** [comment]
I am not sure if we need a different setting for pref and it's, I guess it should be the same. Regarding hotword, a book is just fine.

**Comment by DravitLochan** [comment]
work flow I am looking at :
user login to the account. they see these three options. One option is for android settings, one for iOS and last one for web chat. When clicked on one of them, settings for that client will be displayed.

**Comment by DravitLochan** [comment]
@mariobehling @rishiraj824 Please take a look http://susi-accounts.surge.sh

**Comment by mariobehling** [comment]
Yes, good start. Please proceed. We also need the setting pages in follow up steps.

**Comment by DravitLochan** [comment]
Fixed in #144 Opening issue for  adding settings tab for each one of them.
